### PR TITLE
GFM-26 - Added JSCS and ESLint to the project with rules taken from h…

### PR DESCRIPTION
**Pull request by joefitter**
…ttps://github.com/UKHomeOffice/hof

https://jira.digital.homeoffice.gov.uk/browse/GFM-26

Refactored code in line with code style and linting rules.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+node_modules
+reports
+public/js/bundle.js
+acceptance_tests
+dist/js/bundle.js
+acceptance_tests
+doc/**/*
+apps/**/translations/**/default.json
```
Comment (JoeChapman) [inline .eslintignore]: Not all of these things are in this repo. Remove the items that aren't in the repo

**Comment by JoeChapman** [comment]
I would suggest that we drop support for node v0.12.7 and begin to use ES6 where possible, i.e. arrow functions and `const` etc. However, this all depends on what the dependencies are of this module, @gavboulton, @easternbloc.

Otherwise, LGTM but requires more input from @gavboulton  and @easternbloc

**Comment by easternbloc** [comment]
@joefitter @JoeChapman I'd like us to start using https://github.com/UKHomeOffice/eslint-config-homeoffice to reduce duplication across projects and make upgrades easier. Could you add this to the PR please

**Comment by easternbloc** [comment]
@gavboulton do you have any requirements on older versions of nodejs or are you happy for us to start using es6?

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,15 @@
 language: node_js
 node_js:
-  - "0.10"
-  - "0.12"
-  - "iojs"
+  - "5"
+  - "4"
+  - "4.2"
+
```
Comment (JoeChapman) [inline .travis.yml]: @gavboulton do you foresee this to be a problem?

**Comment by gavboulton** [comment]
We're using an old version of Node at the moment. I'd like to wait until the HO linting module is more mature before using it.

**Comment by easternbloc** [comment]
I think this should be closed till we get it into a position that @gavboulton is happy.

**Pull request closed without merging**